internal/fipcontroller/fipcontroller.go

```diff
@@ -121,11 +124,10 @@ func (fc *Controller) syncFloatingIPs() (bool, error) {
 				fip.Server = srv
 			}
 
-			fc.fips[fip.IP.String()] = fip
+			fc.fips[ip] = fip
 			changedFIPs = true
-		}
-
-		if fip.Server != nil && fip.Server.Name != fc.attachments[fip.IP.String()] {
+		} else if (oldFIP.Server != nil && oldFIP.Server.Name != fc.attachments[ip]) || (oldFIP.Server == nil && fc.attachments[ip] != "") {
```

Comment:
Maybe this would be more readable if we added another little helper:
```go
func fipServerName(fip *hcloud.FloatingIP) string {
	if fip.Server != nil {
		return fip.Server.Name
	}
	return ""
}
```

Then this could become:
```suggestion
		} else if fipServerName(oldFIP) != fc.attachments[ip] {
```

(but don't get me wrong: the original code was already pretty unreadable; that's on me.)

Reply:
Done. Yeah this helper made it way more readable now.